flexmeasures/cli/data_edit.py

```diff
@@ -0,0 +1,241 @@
+    for sensor_id in sensor_ids:
+        sensor = Sensor.query.get(sensor_id)
+        if sensor.event_resolution == event_resolution:
+            print(f"{sensor} already has the desired event resolution.")
+            continue
+        df_original = sensor.search_beliefs(
+            most_recent_beliefs_only=False,
+            event_starts_after=event_starts_after,
+            event_ends_before=event_ends_before,
+        ).sort_values("event_start")
+        df_resampled = df_original.resample_events(event_resolution).sort_values(
+            "event_start"
+        )
+        if not skip_integrity_check:
+            message = ""
+            if sensor.event_resolution < event_resolution:
+                message += f"Downsampling {sensor} to {event_resolution} will result in a loss of data. "
+            click.confirm(
+                message
+                + f"Data before:\n{df_original}\nData after:\n{df_resampled}\nMean before: {df_original['event_value'].mean()}\nMean after: {df_resampled['event_value'].mean()}\nContinue?",
+                abort=True,
+            )
+
+        # Update sensor
+        sensor.event_resolution = event_resolution
+        db.session.add(sensor)
+
+        # Update sensor data
+        query = TimedBelief.query.filter(TimedBelief.sensor == sensor)
+        if not pd.isnull(event_starts_after):
+            query = query.filter(TimedBelief.event_start >= event_starts_after)
+        if not pd.isnull(event_ends_before):
+            query = query.filter(
+                TimedBelief.event_start + sensor.event_resolution <= event_ends_before
+            )
+        query.delete()
+        save_to_db(df_resampled, bulk_save_objects=True)
+    db.session.commit()
+    print("Successfully resampled sensor data.")
+
+
+app.cli.add_command(fm_edit_data)
+
+
+def parse_attribute_value(
+    attribute_null_value: bool,
+    attribute_float_value: Optional[float] = None,
+    attribute_bool_value: Optional[bool] = None,
+    attribute_str_value: Optional[str] = None,
+    attribute_int_value: Optional[int] = None,
+) -> Union[float, int, bool, str, None]:
+    """Parse attribute value."""
+    if not single_true(
+        [attribute_null_value]
+        + [
+            v is not None
+            for v in [
+                attribute_float_value,
+                attribute_bool_value,
+                attribute_str_value,
+                attribute_int_value,
+            ]
+        ]
+    ):
+        raise ValueError("Cannot set multiple values simultaneously.")
+    if attribute_null_value:
+        return None
+    elif attribute_float_value is not None:
+        return float(attribute_float_value)
+    elif attribute_bool_value is not None:
+        return bool(attribute_bool_value)
+    elif attribute_int_value is not None:
+        return int(attribute_int_value)
+    return attribute_str_value
+
+
+def single_true(iterable) -> bool:
+    i = iter(iterable)
+    return any(i) and not any(i)
```

Comment:
I don't get why this would not always return False.

Reply:
[From SO](https://stackoverflow.com/questions/16801322/how-can-i-check-that-a-list-has-one-and-only-one-truthy-value#comment24215913_16801605)...

"`any` will as per docs return True as soon as a non-false value is found. After that, we look for a true value again, and if found treat it as a failure... So this'll work for empty lists, lists/other sequences, and any iterable."

Reply:
Oh that's how `iter` comes in here.